scylla-gdb.py

```diff
@@ -4972,6 +4973,113 @@ def invoke(self, arg, from_tty):
+    [53] 0x60001cce0000
+    [54] 0x60001cd00000
+    [55] 0x60001cd20000
+    [56] 0x60001cd40000
+    [57] 0x60001cd60000
+    [58] 0x60001cd80000
+    [59] 0x60001cda0000
+    """
+
+    def __init__(self):
+        gdb.Command.__init__(self, 'scylla large-objects', gdb.COMMAND_USER, gdb.COMPLETE_COMMAND)
+        self._parser = None
+        self._objects = None
+        self._object_size = None
+
+    def init_parser(self):
+        parser = argparse.ArgumentParser(description="scylla large-objects")
+        parser.add_argument("-o", "--object-size", action="store", type=int, required=True,
+                help="Object size, valid sizes are: ")
+        parser.add_argument("-p", "--page", action="store", type=int, default=0, help="Page to show.")
+        parser.add_argument("-s", "--page-size", action="store", type=int, default=20,
+                help="Number of objects in a page. A page size of 0 turns off paging.")
+        parser.add_argument("--random-page", action="store_true", help="Show a random page.")
+        parser.add_argument("--summarize", action="store_true",
+                help="Print the number of objects and pages in the pool.")
+        parser.add_argument("--verbose", action="store_true",
+                help="Print additional details on what is going on.")
+
+        self._parser = parser
+
+    def invoke(self, arg, from_tty):
+        if self._parser is None:
+            self.init_parser()
+
+        try:
+            args = self._parser.parse_args(arg.split())
+        except SystemExit:
+            return
+
+        object_size = int(args.object_size / int(gdb.parse_and_eval('\'seastar::memory::page_size\'')))
+
+        if self._objects is None or object_size != self._object_size:
+            if args.verbose:
+                gdb.write(f"Object size changed ({self._object_size} pages -> {object_size} pages), scanning spans.\n")
+            # We materialize all relevant spans. There aren't usually that many
+            # of them. Rewrite this to be lazy if this proves problematic with
+            # certain cores.
+            self._objects = [s for s in spans() if s.is_large() and not s.is_free() and s.size() == object_size]
+
+        self._object_size = object_size
+
+        if args.summarize:
+            gdb.write("number of objects: {}\n"
+                      "page size        : {}\n"
+                      "number of pages  : {}\n"
+                      .format(
+                          len(self._objects),
+                          args.page_size,
+                          math.ceil(len(self._objects) / args.page_size)))
+            return
+
+        if args.random_page:
+            page = random.randint(0, int(len(self._objects) / args.page_size) - 1)
+        else:
+            page = args.page
+
+        if page >= len(self._objects)/args.page_size:
+            return
+
+        start = page * args.page_size
+        end = min(start + args.page_size, len(self._objects))
```

Comment:
nit, could just put 
```py
end = start + args.page_size
```

as per https://docs.python.org/3/library/stdtypes.html#typesseq

> The slice of s from i to j is defined as the sequence of items with index k such that `i <= k < j`. If i or j is greater than `len(s)`, use `len(s)`

Reply:
Done in v2.